system/security/shared/caching.cpp

```diff
@@ -672,16 +680,55 @@ bool CPermissionsCache::queryPermsManagedFileScope(ISecUser& sec_user, const cha
 
 SecAccessFlags CPermissionsCache::queryDefaultPermission(ISecUser& user)
 {
-    if (m_defaultPermission == SecAccess_Unknown)
+
+    if (m_useLegacyDefaultFileScopePermissionCache)
     {
-        if (m_secMgr)
-            m_defaultPermission = m_secMgr->queryDefaultPermission(user);
+        if (m_defaultPermission == SecAccess_Unknown)
+        {
+            if (m_secMgr)
+                m_defaultPermission = m_secMgr->queryDefaultPermission(user);
+            else
+                m_defaultPermission = SecAccess_Full;
```

Comment:
This is different from the code in 9.2.x/9.4.x.  Is this deliberate?  I suspect it is, but could do with an indication why.

Reply:
A previous outdated comment indicates why it is set to Full. If there is no security manager, access should be full for all users and all scopes. However, technically as currently implemented, this code will never be called if there is not security manager. However, must still handle the case. I will a short comment on that line.

Reply:
The case where the legacy cache is not used still returns SecAccess_None.  It be clearer to test for m_secMgr at the start of the function and return SecAccess_Full.